src/components/MenuPanel/MenuPanelAppsLoader.js

```diff
@@ -1,78 +1,134 @@
 import React from 'react'
 import styled from 'styled-components'
-import { spring, Motion } from 'react-motion'
-import { spring as springConf } from '@aragon/ui'
+import { Spring, animated } from 'react-spring'
+import { IconAttention, Text } from '@aragon/ui'
 import { lerp } from '../../math-utils'
+import { noop } from '../../utils'
 import LoadingRing from '../LoadingRing'
-
-const SPRING = springConf('fast')
+import {
+  APPS_STATUS_ERROR,
+  APPS_STATUS_READY,
+  APPS_STATUS_LOADING,
+} from '../../symbols'
 
 class MenuPanelAppsLoader extends React.Component {
-  state = { hideLoader: false, loadingTransitionDone: false }
-
-  componentWillReceiveProps({ loading }) {
-    if (loading && !this.props.loading) {
-      this.setState({ hideLoader: false, loadingTransitionDone: false })
-    }
+  static defaultProps = {
+    children: noop,
+    itemsCount: 0,
+    appStatus: APPS_STATUS_LOADING,
+    onRetry: noop,
   }
 
-  handleRest = () => {
-    const { loading } = this.props
+  state = {
+    showApps: false,
+  }
 
-    if (loading) {
-      return
+  componentWillReceiveProps({ appsStatus }) {
+    if (appsStatus !== this.props.appsStatus) {
+      this.setState({ showApps: false })
     }
+  }
 
-    if (!this.state.hideLoader) {
-      // To trigger the Motion transition
+  handleRest = () => {
+    if (this.props.appsStatus === APPS_STATUS_READY) {
       setTimeout(() => {
-        this.setState({ hideLoader: true })
+        this.setState({ showApps: true })
       }, 0)
-    } else {
-      this.setState({ loadingTransitionDone: true })
     }
   }
+
   render() {
-    const { children, itemsCount, loading } = this.props
-    const { hideLoader, loadingTransitionDone } = this.state
+    const { children, itemsCount, appsStatus } = this.props
+    const { showApps } = this.state
     return (
-      <Motion
-        onRest={this.handleRest}
-        defaultStyle={{ prepareProgress: 0, hideProgress: 0 }}
-        style={{
-          prepareProgress: spring(Number(!loading), SPRING),
-          hideProgress: spring(Number(hideLoader), SPRING),
+      <Spring
+        config={{ tension: 800, friction: 60 }}
```

Comment:
Should we add this config somewhere, or is it pretty custom?

Reply:
Yes we should move it to Aragon UI, I would like to make some tests to see if we update `swift` or if we need a new one.